lib/logstash/outputs/google_bigquery.rb

```diff
@@ -274,11 +276,21 @@ def write_to_errors_file(messages, table)
 
   def init_batcher_flush_thread
     @flush_thread = Thread.new do
-      loop do
-        sleep @flush_interval_secs
+      until stopping?
```

Comment:
I can see a narrow race here, in which the pipeline receives a shutdown signal after this output has begun flushing, which would cause it to not enter this loop again and therefore not receive a final flush.

Reply:
Good catch @yaauie

Reply:
🦅 eyes